lib/action_subscriber/publisher/async.rb

```diff
@@ -0,0 +1,31 @@
+module ActionSubscriber
+  module Publisher
+    # Publish a message asynchronously to RabbitMQ.
+    #
+    # Asynchronous is designed to do two things:
+    # 1. Introduce the idea of a durable retry should the RabbitMQ connection disconnect.
+    # 2. Provide a higher-level pattern for fire-and-forget publishing.
+    #
+    # @param [String] route The routing key to use for this message.
+    # @param [String] payload The message you are sending. Should already be encoded as a string.
+    # @param [String] exchange The exchange you want to publish to.
+    # @param [Hash] options hash to set message parameters (e.g. headers).
+    def self.publish_async(route, payload, exchange_name, options = {})
+      Async.publisher_adapter.publish(route, payload, exchange_name, options)
+    end
+
+    module Async
+      def self.publisher_adapter
+        @publisher_adapter ||= case ::ActionSubscriber.configuration.async_message_publisher_mode
+                               when /memory/i then
```

Comment:
Why use a regex here? It seems reasonable that a lowercase string would need to be used in the config. Alternatively, you could require a symbol.